notebooks/tests/test_config.py

```diff
@@ -0,0 +1,105 @@
+import yaml
+from pytest import raises
+from testbook import testbook
+from testbook.exceptions import TestbookRuntimeError
+from uwtools.exceptions import UWConfigError, UWConfigRealizeError
+
+all_cells = [*range(0, 57)]
+err_cells = [13, 17, 45]
+non_err_cells = list(filter(lambda n: n not in err_cells, all_cells))
+
+
+def test_get_config():
+    with open("fixtures/config/get-config.yaml", "r", encoding="utf-8") as f:
+        config1_str = f.read().rstrip()
+        config1_dict = yaml.safe_load(config1_str)
+    with testbook("config.ipynb", execute=non_err_cells) as tb:
+        assert tb.ref("config1") == config1_dict
+        assert tb.cell_output_text(5) == config1_str
+        assert tb.cell_output_text(7) == config1_str
+        assert tb.cell_output_text(9) == "message:\n  greeting: Hi\n  recipient: Earth"
+
+
+def test_depth():
+    with testbook("config.ipynb", execute=non_err_cells) as tb:
+        assert tb.cell_output_text(11) == "greeting=Salutations\nrecipient=Mars"
+        assert tb.cell_output_text(15) == "[message]\ngreeting = Salutations\nrecipient = Mars"
+
+
+def test_depth_errs():
+    with raises(TestbookRuntimeError) as e:
+        with testbook("config.ipynb", execute=[1, 13]):
+            pass
+    assert e.value.eclass == UWConfigError
+    assert e.value.evalue == "Cannot instantiate depth-1 SHConfig with depth-2 config"
+    with raises(TestbookRuntimeError) as e:
+        with testbook("config.ipynb", execute=[1, 17]):
+            pass
+    assert e.value.eclass == UWConfigError
+    assert e.value.evalue == "Cannot instantiate depth-2 INIConfig with depth-1 config"
+
+
+def test_realize():
+    # Get config file data to compare to cell output.
+    with open("fixtures/config/get-config.yaml", "r", encoding="utf-8") as f:
+        config_str = f.read().rstrip()
+        config_dict = yaml.safe_load(config_str)
+    with open("fixtures/config/base-config.nml", "r", encoding="utf-8") as f:
+        update_config_str = f.read().rstrip()
+    with open("fixtures/config/keys-config.yaml", "r", encoding="utf-8") as f:
+        keys_config_str = f.read().rstrip()
+    with testbook("config.ipynb", execute=non_err_cells) as tb:
+        # Ensure that cell output text matches expectations.
+        assert tb.cell_output_text(21) == str(config_dict)
+        assert tb.cell_output_text(23) == config_str
+        assert tb.cell_output_text(25) == str(config_dict)
```

Comment:
If you loaded the config files with the UW API and named them `config` for the purposes of this , the object you get back can be modified to match these tests.

For the string comparison, you should be able to go with `assert tb.cell_output_text(21) == str(config)`, and for the dict comparison, convert a UW Config object to a `dict`, so the test looks like `assert tb.cell_output_text(25) == str(dict(cfg))`.